docs/app/AuthenticationPage.tsx

```diff
@@ -0,0 +1,359 @@
+          {signingInState.message}
+        </p>
+      )}
+    </form>
+  );
+}
+
+function AlternativeSignInButton(props: {
+  name: string;
+  icon: ReactNode;
+  onClick: () => void;
+}) {
+  return (
+    <button
+      className="flex w-full items-center justify-center gap-3 rounded-md bg-white px-3 py-2 text-sm font-semibold text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 hover:bg-gray-50 focus-visible:ring-transparent dark:bg-gray-800 dark:text-gray-100 dark:ring-gray-700 dark:hover:bg-gray-700"
+      onClick={props.onClick}>
+      {props.icon}
+      <span className="text-sm/6 font-semibold">{props.name}</span>
+    </button>
+  );
+}
+
+function EmailSignInButton() {
+  const router = useRouter();
+
+  return (
+    <AlternativeSignInButton
+      name="Email"
+      icon={
+        <svg
+          fill="currentColor"
+          viewBox="0 -960 960 960"
+          aria-hidden="true"
+          className="size-5">
+          <path d="M160-160q-33 0-56.5-23.5T80-240v-480q0-33 23.5-56.5T160-800h640q33 0 56.5 23.5T880-720v480q0 33-23.5 56.5T800-160H160Zm320-280 320-200v-80L480-520 160-720v80l320 200Z" />
+        </svg>
+      }
+      onClick={() => {
+        const searchParams = new URLSearchParams(window.location.search);
+        const callbackURL = searchParams.get("redirect") || "/";
+        const theme = searchParams.get("theme") || "/system";
+
+        router.push(`/signin/email?redirect=${callbackURL}&theme=${theme}`);
+      }}
+    />
+  );
+}
+
+function GitHubSignInButton() {
+  return (
+    <AlternativeSignInButton
+      name="GitHub"
+      icon={
+        <svg
+          fill="currentColor"
+          viewBox="0 0 20 20"
+          aria-hidden="true"
+          className="size-5">
+          <path
+            d="M10 0C4.477 0 0 4.484 0 10.017c0 4.425 2.865 8.18 6.839 9.504.5.092.682-.217.682-.483 0-.237-.008-.868-.013-1.703-2.782.605-3.369-1.343-3.369-1.343-.454-1.158-1.11-1.466-1.11-1.466-.908-.62.069-.608.069-.608 1.003.07 1.531 1.032 1.531 1.032.892 1.53 2.341 1.088 2.91.832.092-.647.35-1.088.636-1.338-2.22-.253-4.555-1.113-4.555-4.951 0-1.093.39-1.988 1.029-2.688-.103-.253-.446-1.272.098-2.65 0 0 .84-.27 2.75 1.026A9.564 9.564 0 0110 4.844c.85.004 1.705.115 2.504.337 1.909-1.296 2.747-1.027 2.747-1.027.546 1.379.203 2.398.1 2.651.64.7 1.028 1.595 1.028 2.688 0 3.848-2.339 4.695-4.566 4.942.359.31.678.921.678 1.856 0 1.338-.012 2.419-.012 2.747 0 .268.18.58.688.482A10.019 10.019 0 0020 10.017C20 4.484 15.522 0 10 0z"
+            clipRule="evenodd"
+            fillRule="evenodd"
+          />
+        </svg>
+      }
+      onClick={async () => {
+        const searchParams = new URLSearchParams(window.location.search);
+        const callbackURL = searchParams.get("redirect") || "/";
+
+        await signIn.social({
+          provider: "github",
+          callbackURL,
+        });
+      }}
+    />
+  );
+}
+
+function AlternativeSignInBox(props: {
```

Comment:
I think I like how supabase did it here better: 
<img width="504" alt="image" src="https://github.com/user-attachments/assets/1e4be979-2f93-4448-80b6-b0440ffc230f" />

Where the button specifically says "continue with github" and "continue with email"

Just a comment and minor preference